source/Halibut/Transport/Protocol/WebSocketStream.cs

```diff
@@ -37,47 +40,74 @@ public override void SetLength(long value)
         public override int Read(byte[] buffer, int offset, int count)
         {
             AssertCanReadOrWrite();
-            var segment = new ArraySegment<byte>(buffer, offset, count);
-            var receiveResult = context.ReceiveAsync(segment, CancellationToken.None)
+            return ReadAsync(buffer, offset, count, CancellationToken.None)
```

Comment:
Just avoid changing the existing implementations, but instead add the Async or 'Really'Async methods